test/Dialect/FIRRTL/parse-name-preservation.fir

```diff
@@ -0,0 +1,22 @@
+; RUN: firtool --parse-only %s | FileCheck %s
+circuit Bar :
+  ; NamePreservation should create attaches for analog typed elements.
+  ; https://github.com/llvm/circt/issues/2718
+  module Bar :
+    input a: {flip a: UInt<1>, b: Analog<1>}[1]
+    output b: {flip a: UInt<1>, b: Analog<1>}[1]
+
+    wire w: {flip a: UInt<1>, b: Analog<1>}[1]
+    ; CHECK: %_w = firrtl.wire
```

Comment:
Checking the type of the renamed wire would be good, as this needs to be the same as the original FIRRTL source.

Reply:
I added some more thorough testing over here: https://github.com/llvm/circt/pull/2724